domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/Attribution.java

```diff
@@ -0,0 +1,397 @@
+                        && (originalIsOperatorInteger < 0 || originalIsOperatorInteger > 1)) {
+                    noticeCollection.add(new IntegerFieldValueOutOfRangeNotice("attributions.txt",
+                            "is_operator", entityId, 0, 1, originalIsOperatorInteger));
+                }
+                if (originalIsAuthorityInteger != null
+                        && (originalIsAuthorityInteger < 0 || originalIsAuthorityInteger > 1)) {
+                    noticeCollection.add(new IntegerFieldValueOutOfRangeNotice("attributions.txt",
+                            "is_authority", entityId, 0, 1, originalIsAuthorityInteger));
+                }
+                if (originalIsProducerInteger != null
+                        && (originalIsProducerInteger < 0 || originalIsProducerInteger > 1)) {
+                    noticeCollection.add(new IntegerFieldValueOutOfRangeNotice("attributions.txt",
+                            "is_producer", entityId, 0, 1, originalIsProducerInteger));
+                }
+                if ((isAuthority == isProducer) && (isAuthority == isOperator) &&
+                        (originalIsProducerInteger == null || originalIsProducerInteger == 0)) {
+                    noticeCollection.add(
+                            new IllegalFieldValueCombination("attributions.txt", "is_producer",
+                                    "is_authority; is_operator",
+                                    entityId));
+                }
+                return new EntityBuildResult<>(noticeCollection);
+            } else {
+                return new EntityBuildResult<>(new Attribution(attributionId, agencyId, routeId, tripId,
+                        organizationName, isProducer, isOperator, isAuthority, attributionUrl, attributionEmail,
+                        attributionPhone));
+            }
+        }
+    }
+
+    /**
+     * Returns the key used to map {@code Attribution}
+     *
+     * @return  the key used to map {@link Attribution}
+     */
+    public static String getAttributionMappingKey(final String attributionId, final String agencyId,
+                                                  final String routeId, final String tripId,
+                                                  final String organizationName, final Boolean isProducer,
+                                                  final Boolean isOperator, final Boolean isAuthority,
+                                                  final String attributionUrl, final String attributionEmail,
+                                                  final String attributionPhone) {
+        return attributionId+agencyId+routeId+tripId+organizationName+isProducer+isOperator+isAuthority+attributionUrl+
+                attributionEmail+attributionPhone;
+    }
+
+    /**
+     * Returns the key corresponding to this {@link Attribution}
+     *
+     * @return the key corresponding to this {@link Attribution}
+     */
+    public String getAttributionMappingKey() {
```

Comment:
I seem to recognize the pattern that led to code duplication. This should be implemented through a static method

Reply:
The static method is implemented at line 377 of this same file, as done in FareRule.java

Reply:
Very true, my bad. thks for the quick follow up.